### PR TITLE
prow: pod-utils: sidecar: ignore SIGINT, SIGTERM

### PR DESCRIPTION
**Pull request by stevekuznetsov**
There is no valid way for the sidecar to exit before it has finished
uploading all artifacts, which can only occur once the test process has
exited. We therefore ignore SIGINT and SIGTERM and hope that we can get
our work done before the kubelet sends us SIGKILL.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/area prow
/kind bug
/cc @BenTheElder @fejta @krzyzacy 
/assign @cjwagner

### CODE REVIEW

File: prow/sidecar/run.go [hunk 1]
```diff
@@ -38,6 +40,13 @@ import (
 // and then post the status of that process and any artifacts
 // to cloud storage.
 func (o Options) Run() error {
+	// If we are being asked to terminate by the kubelet we must
+	// hope that the test process exits cleanly and we can get a
+	// chance to upload artifacts to GCS. The only valid way for
+	// this program to exit after a SIGINT or SIGTERM is to finish
+	// uploading, so we ignore the signal.
+	signal.Ignore(os.Interrupt, syscall.SIGTERM)
```
Comment (BenTheElder) [inline prow/sidecar/run.go]: could we not opportunistically start uploading whatever artifacts we have upon receiving these signals?

Comment (stevekuznetsov) [reply prow/sidecar/run.go]: What should we do if we upload some partial result (or, for instance, scan a directory as empty) at `SIGTERM` but then we observe the test process exit? Cancel all pending uploads and rescan?

Comment (BenTheElder) [reply prow/sidecar/run.go]: bandwidth isn't usually a concern for these I think (?), so my first thought is that as long as we upload in parallel we can rescan and upload [on process exit] without cancelling the previous ones [from SIGTERM]. 
sure it's racy, but if we cancel the upload we might not finish posting anything, where as if the rescan finishes uploading I think the likely scenario is those uploads [after process exit] will override the SIGTERM uploads. 
and GCS will only replace an already uploaded entry ~ atomically if we actually mark the upload complete, so if the rescan starts to upload something but never finishes we hopefully still finished the one from SIGTERM.

I definitely think it's more useful to guarantee we upload whatever we can when pods are less-than-gracefully terminated than nothing.

If we observe SIGTERM after we've already observed process exit somehow, we should not do anything of course.

**Comment by BenTheElder** [comment]
LGTM too, thanks!

**Pull request merged**